### PR TITLE
Minor API changes to RecordReader

### PR DESCRIPTION
**Pull request by raver119**
Minor API change to RecordReader:
- batchesSupported() 
- next(num)

This method provides List<Writable> that contains whole batch at once, so DataSet.merge() call isn't needed

### CODE REVIEW

File: datavec-data/datavec-data-image/src/main/java/org/datavec/image/recordreader/BaseImageRecordReader.java [hunk 1]
```diff
@@ -250,6 +254,56 @@ public boolean hasNext() {
...
+        List<Integer> currLabels = new ArrayList<>();
+        while (cnt < num && iter.hasNext()) {
+            File currFile = iter.next();
+            currBatch.add(currFile);
+            if (appendLabel)
+                currLabels.add(labels.indexOf(getLabel(currFile.getPath())));
+            cnt++;
+        }
+
+        INDArray features = Nd4j.createUninitialized(new int[]{cnt, channels, height, width});
```
Comment (AlexDBlack) [inline datavec-data/datavec-data-image/src/main/java/org/datavec/image/recordreader/BaseImageRecordReader.java]: Probably doesn't matter give the default ordering, but maybe this should be a createUninitialized with C order (for strides)?

File: datavec-api/src/main/java/org/datavec/api/records/reader/RecordReader.java [hunk 2]
```diff
@@ -64,13 +64,29 @@
...
+     */
+    boolean batchesSupported();
+
+    /**
+     * This method will be used, if batchesSupported() returns true.
+     *
+     * @param num
+     * @return
+     */
+    List<Writable> next(int num);
```
Comment (AlexDBlack) [inline datavec-api/src/main/java/org/datavec/api/records/reader/RecordReader.java]: I'm not *totally* happy with this new API, as it's previously always been the case that that a ```List<Writable>``` is a single example. Now it's somewhat opaque to the number of examples in it. Longer term I'd like to review this and maybe replace it with an object called something like ```BatchRecord```.

Comment (raver119) [inline datavec-api/src/main/java/org/datavec/api/records/reader/RecordReader.java]: Totally agree... I wasn't going to merge this, but then varts bug was found :(
I'm still working on parallel etl, so i'll change it in next PR

File: datavec-data/datavec-data-image/src/main/java/org/datavec/image/recordreader/BaseImageRecordReader.java (hunk 1 above)
Comment (raver119) [inline datavec-data/datavec-data-image/src/main/java/org/datavec/image/recordreader/BaseImageRecordReader.java]: Agreed.

File: datavec-api/src/main/java/org/datavec/api/records/reader/RecordReader.java (hunk 2 above)
Comment (raver119) [inline datavec-api/src/main/java/org/datavec/api/records/reader/RecordReader.java]: I still need the same for CSV stuff anyway.

File: datavec-dataframe/pom.xml [hunk 3]
```diff
@@ -114,6 +114,12 @@
             <artifactId>jfairy</artifactId>
             <version>0.5.2</version>
             <scope>test</scope>
+            <exclusions>
+                <exclusion>
+                    <groupId>com.google.guava</groupId>
+                    <artifactId>guava</artifactId>
```
Comment (benmccann) [inline datavec-dataframe/pom.xml]: why exclude guava?

**Pull request merged**